pkg/controllers/job/job_controller_actions.go

```diff
@@ -465,6 +470,11 @@ func (cc *jobcontroller) syncJob(jobInfo *apis.JobInfo, updateStatus state.Updat
 		RetryCount:          job.Status.RetryCount,
 	}
 
+	if reflect.DeepEqual(newStatus, job.Status) {
```

Comment:
We should call `updateStatus` first before compare.

Reply:
ok. done